econml/inference.py

```diff
@@ -45,26 +45,69 @@ class BootstrapInference(Inference):
     n_jobs: int, optional (default -1)
         The maximum number of concurrently running jobs, as in joblib.Parallel.
 
+    bootstrap_type: 'percentile', 'pivot', or 'normal', default 'percentile'
+        Bootstrap method used to compute results.
+        'percentile' will result in using the empiracal CDF of the replicated copmutations of the statistics.
+        'pivot' will also use the replicates but create a pivot interval that also relies on the estimate
+        over the entire dataset.
+        'normal' will instead compute a pivot interval assuming the replicates are normally distributed.
     """
 
-    def __init__(self, n_bootstrap_samples=100, n_jobs=-1):
+    def __init__(self, n_bootstrap_samples=100, n_jobs=-1, bootstrap_type='percentile'):
         self._n_bootstrap_samples = n_bootstrap_samples
         self._n_jobs = n_jobs
+        self._bootstrap_type = bootstrap_type
 
     def fit(self, estimator, *args, **kwargs):
-        est = BootstrapEstimator(estimator, self._n_bootstrap_samples, self._n_jobs, compute_means=False)
+        discrete_treatment = estimator._discrete_treatment if hasattr(estimator, '_discrete_treatment') else False
+        est = BootstrapEstimator(estimator, self._n_bootstrap_samples, self._n_jobs, compute_means=False,
+                                 bootstrap_type=self._bootstrap_type)
         est.fit(*args, **kwargs)
         self._est = est
+        self._d_t = estimator._d_t
+        self._d_y = estimator._d_y
+        self.d_t = self._d_t[0] if self._d_t else 1
+        self.d_y = self._d_y[0] if self._d_y else 1
 
     def __getattr__(self, name):
         if name.startswith('__'):
             raise AttributeError()
 
         m = getattr(self._est, name)
+        if name.endswith('_interval'):  # convert alpha to lower/upper
+            def wrapped(*args, alpha=0.1, **kwargs):
+                return m(*args, lower=100 * alpha / 2, upper=100 * (1 - alpha / 2), **kwargs)
+            return wrapped
+        else:
+            return m
 
-        def wrapped(*args, alpha=0.1, **kwargs):
-            return m(*args, lower=100 * alpha / 2, upper=100 * (1 - alpha / 2), **kwargs)
-        return wrapped
+    def summary(self, alpha=0.1, value=0, decimals=3, feat_name=None):
```

Comment:
Isn't this piece of code already implemented in LinearModelFinalInference?

Reply:
Fixed by adding a mixin that both classes use.